chromium_src/content/browser/tld_ephemeral_lifetime.cc

```diff
@@ -29,10 +30,11 @@ TLDEphemeralLifetimeMap& active_tld_storage_areas() {
 
 TLDEphemeralLifetime::TLDEphemeralLifetime(TLDEphemeralLifetimeKey key,
                                            StoragePartition* storage_partition)
-    : key_(key), storage_partition_(storage_partition) {
-  DCHECK(active_tld_storage_areas().find(key) ==
+    : key_(std::move(key)), storage_partition_(storage_partition) {
```

Comment:
the reasoning for changes in this file: the implementation was somewhere in the middle - it was accepting parameters by value but didn't ```move``` them internally. We should use ```move``` or the interface should be ```const&```, I decided to go with the first.

Everytime we need a copy of the value (which is 99% constructors do), it's just clearer and more efficient to receive it by value:
* ```const&``` impl will always do a copy
* ```by value``` impl will do a copy only if a caller didn't move/pass rvalue (i.e., same as ```const&```, otherwise better)

> I don't think the minor efficiency of skipping one copy for a small object is worth using this pattern which is otherwise unconventional

I agree about minor efficiency in any single case, but you should look more widely imo. Every unnecessary copy on each browser loop can be crucial on slow/small core CPUs. Startup performance, power usage, etc. can be affected in the long term.

Reply:
Yeah, pure `std::string` in the method interface implies that callers should rather move than copy, this is a common convention. 

the downside of not using const ref in interfaces is that people tend to forget about std::move (or even don't know that they should use it), as we see in this code.

there is also an old but interesting thread re strings performance https://groups.google.com/a/chromium.org/d/msg/chromium-dev/EUqoIz2iFU4/kPZ5ZK0K3gEJ

Reply:
replaced movable args with ```const&``` in ```TLDEphemeralLifetime```.